libzeth/core/bits.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
+//
+// SPDX-License-Identifier: LGPL-3.0+
+
+#include "libzeth/core/bits.hpp"
+
```

Comment:
Extraneous line?

Reply:
I'm not sure why, but clang-format insists on putting thsi extra line in.  I'm not sure sure it happens here and nowhere else, but I can't find a way to avoid it.

Reply:
TBH, I didn't look in detail (just tried removing it, but it always gets put back).  My guess is that clang-format groups the headers as "local header" "includes from local library" "external includes", with a space between each.

Reply:
I guess clang doesn't apply this rule in `.tcc` files, as it doesn't consider them as "source" files.

Reply:
> I guess clang doesn't apply this rule in .tcc files, as it doesn't consider them as "source" files.

Not sure I get the point. This file is a `cpp` file, not a `tcc` file, right? And both includes are `libzeth/core/*` so they should both be "local headers", no?

Reply:
> Not sure I get the point. This file is a cpp file, not a tcc file, right? And both includes are libzeth/core/* so they should both be "local headers", no?

Since `.tcc` is not really a well known file extension, I assume clang-format doesn't recognise the "local" include of `bits.hpp` in thi file, so `bits.hpp` gets grouped with all othe rincludes from the library.  In `.cpp` files, clang-format understands the file extension and therefore *does* recognise local headers, hence this separation between `bits.hpp` and other headers.